src/region.js

```diff
@@ -315,42 +236,39 @@ var Region = MarionetteObject.extend({
     if (view.destroy) {
       view.destroy();
     } else {
+      const shouldTriggerDetach = !!view._isAttached;
```

Comment:
I actually did one better and moved all "removal" logic to `_removeView`, which is only called if view isn't yet destroyed.

```js
  _removeView(view, {preventDestroy} = {}) {
    const shouldPreventDestroy = !!preventDestroy;

    if (shouldPreventDestroy) {
      this._detachView(view);
      return;
    }

    if (view.destroy) {
      view.destroy();
    } else {
      destroyBackboneView(view);
    }
  },
```